game/src/ltn/algorithms.rs

```diff
@@ -0,0 +1,146 @@
+use std::collections::{BTreeSet, HashMap};
+
+use geom::Distance;
+use map_model::osm::RoadRank;
+use map_model::{IntersectionID, Map, PathConstraints, RoadID};
+
+use crate::ltn::{Neighborhood, RatRun};
+
+impl Neighborhood {
+    // TODO Doesn't find the full perimeter. But do we really need that?
+    pub fn from_road(map: &Map, start: RoadID) -> Neighborhood {
```

Comment:
When you click on a local road, this floodfills out to all the major roads, auto-discovering some approximation of the neighborhood. There are some issues with existing modal filters mapped as cycle-only roads in OSM. And as Michael brought up, this kind of automation can be useful, but it shouldn't get in the way of somebody drawing their area of interest manually.

Reply:
This is such a great feature and, as far as I know, it's unique. Great invention!